src/main/java/com/github/trex_paxos/srs/FileRecordStore.java

```diff
@@ -230,6 +230,25 @@ public Iterable<ByteSequence> keys() {
         return snapshot.stream().map(ByteSequence::copy).collect(Collectors.toSet());
     }
 
+    /// Captures the on-disk index ordering so crash tests can validate
+    /// structural invariants without exposing mutable headers.
+    @Synchronized
+    List<RecordSnapshot> snapshotRecords() {
```

Comment:
i do not like the use of List here we can move to flattened value objects in some future java so make this a array then use Arrays.setAll and the like